File: LP2/Shirt.php
<?php

/*Saya Muhammad Alfi faiz NIM 2207045 mengerjakan
soal Latihan 2 dalam mata kuliah Desain Pemograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

require_once 'Clothing.php';

class Shirt extends Clothing {
    private $warna;
    private $jenis_lengan;

    public function __construct($id_produk, $nama, $merek, $harga, $ukuran, $material, $jenis_kelamin, $warna, $jenis_lengan) {
        parent::__construct($id_produk, $nama, $merek, $harga, $ukuran, $material, $jenis_kelamin);
        $this->warna = $warna;
        $this->jenis_lengan = $jenis_lengan;
    }

    public function get_Warna() {
        return $this->warna;
    }

    public function set_Warna($warna) {
        $this->warna = $warna;
    }

    public function getS_tipe_lengan() {
        return $this->jenis_lengan;
    }

    public function setS_tipe_lengan($jenis_lengan) {
        $this->jenis_lengan = $jenis_lengan;
    }
}

?>

File: CPP/Shirt.cpp

/*Saya Muhammad Alfi faiz NIM 2207045 mengerjakan
soal Latihan 2 dalam mata kuliah Desain Pemograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

#include <iostream>
#include <vector>
#include <string>
#include "Clothing.cpp"
using namespace std;

class Shirt : public Clothing {
private:
    string warna;
    string jenis_lengan;

public:
    Shirt(int id_produk, string nama, string merek, double harga, string ukuran, string material, string jenis_kelamin, string warna, string jenis_lengan)
        : Clothing(id_produk, nama, merek, harga, ukuran, material, jenis_kelamin), warna(warna), jenis_lengan(jenis_lengan) {}

    string get_Warna() const {
        return warna;
    }

    void set_Warna(string warna) {
        this->warna = warna;
    }

    string getS_tipe_lengan() const {
        return jenis_lengan;
    }

    void setS_tipe_lengan(string jenis_lengan) {
        this->jenis_lengan = jenis_lengan;
    }
};